modules/navier_stokes/src/fvkernels/INSFVBodyForce.C

```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVBodyForce.h"
+#include "Function.h"
+
+registerMooseObject("NavierStokesApp", INSFVBodyForce);
+
+InputParameters
+INSFVBodyForce::validParams()
+{
+  auto params = INSFVElementalKernel::validParams();
+  params.addClassDescription("Body force that contributes to the Rhie-Chow interpolation");
+  params.addParam<Real>("value", 1.0, "Coefficient to multiply by the body force term");
+  params.addParam<FunctionName>("function", "1", "A function that describes the body force");
```

Comment:
let s use a functor here?
That way no INSFVMatBodyForce or INSFVCoupledForce needed

Reply:
Can it be a `Real` type? I don't want to go and update all the MMS inputs 😢

Reply:
For MMS they are all parsed functions, not simple constants. But it's fine, there's not that many inputs. I'll just go change it to an ADReal functor. Zero-size AD arithmetic should not be that expensive either in the case that the functor is a constant with respect to the nonlinear degrees of freedom